gimbal-adapter/src/main/java/com/urbanairship/gimbal/GimbalAdapter.java

```diff
@@ -284,12 +285,22 @@ private void startAdapter(@NonNull String gimbalApiKey) {
      * Stops the adapter.
      */
     public void stop() {
+        if (!isStarted()) {
+            Log.w(TAG, "stop() called when adapter was not started");
+            return;
+        }
+
         if (requestPermissionsTask != null) {
             requestPermissionsTask.cancel(true);
         }
 
-        Gimbal.stop();
-        PlaceManager.getInstance().removeListener(placeEventListener);
+        try {
+            Gimbal.stop();
+            PlaceManager.getInstance().removeListener(placeEventListener);
+        } catch (Exception e) {
```

Comment:
I think that unless there are other known exceptions that Gimbal.stop can throw, if this change fixes the NPE we can do without a blanket try/catch.

Reply:
Agreed.